test/scripts/setup_sources.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env sh
+
+# Generate a tarball from the given files and update the sha256 in the opam file
+
+set -eu
+
```

Comment:
I suggest also adding

```suggestion
set -eu
set -o pipefail
```

Reply:
Dash doesn't seem to have `pipefail`, but `bash` does. I'm not sure if we should just switch to running the script with `bash`, given all the OSes and architectures we run the CI on. I'm leaning towards making this a follow-up, and merging this PR for now.

Reply:
Oh I misssed this was dash! I'm not familiar with it and maybe it doesn't need pipefail. No worries here :) review oversight in my part.